h/streamer/websocket.py

```diff
@@ -18,8 +18,28 @@
 # below.
 MESSAGE_HANDLERS = {}
 
+
 # An incoming message from a WebSocket client.
-Message = namedtuple('Message', ['socket', 'payload'])
+class Message(namedtuple('Message', [
+    'socket',
+    'payload',
+])):
+    def reply(self, payload, ok=True):
+        """
+        Send a response to this message.
+
+        Sends a reply message back to the client, with the passed `payload`
+        and reporting status `ok`.
+        """
+        reply_to = self.payload.get('id')
+        # Short-circuit if message is missing an ID or has a non-numeric ID.
+        if not isinstance(reply_to, (int, float)):
+            return
```

Comment:
It probably outweighs the drawbacks because the goal is to change the client and to get rid of the old websocket API. But this code would be nicer if it would raise an error and the caller has to make sure to only reply when there's an id in the request. The way the code is written now will probably introduce some situations where it's hard to debug why the client isn't getting a reply. I would imagine that once we remove the old API we would immediately error / close when we get a request without an id, without even trying to do anything on the server.